.github/workflows/sematic-pr-title.yml

```diff
@@ -0,0 +1,55 @@
+name: "Sematic PR Title"
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: amannn/action-semantic-pull-request@v3.4.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          types: |
+            build
+            ci
+            fix
+            feat
+            refactor
+            chore
+            docs
+            style
+            perf
+            test
+            i18n
```

Comment:
I think this is the part I'm least sure about.  If other folks want this, I'm happy to go for it with some minor tweaks, but if it was just up to me I would be not sure.

My feelings and thoughts here are:
* The nicest part of this is having a bot being able to enforce commit message structure.
* This list as written has lots of overlap in `types` and `scopes`.  It seems like build, ci, docs, style, test, i18n are all `scopes` and not `types`?
* I don't find a lot of value in knowing `fix` vs `feat`.  I think it's often pretty ambiguous (features can exist to "fix" old/missing code?)
* This feels slightly redundant with https://chris.beams.io/posts/git-commit/#imperative, where I find that a good PR title has an imperative verb to start (e.g. add/fix/refactor/remove) and so `fix(raidboss): fix thing` or `feat(oopsy): add thing` don't seem to add much more than `raidboss: fix thing` or `oopsy: add thing` and it takes up space in an already space-limited PR title
* I know we have PR labels, but I do think `scopes` is still useful for browsing commit messages, as the labels don't appear in `git log`.
* This is maybe a larger issue, but I am definitely not following semantic versioning for cactbot, mostly because I have never figured out what to really consider a breaking change (happy to discuss if people want to define this better?), and so semantic PR titles have less value because we're not doing semantic versioning

Reply:
> * This list as written has lots of overlap in `types` and `scopes`.  It seems like build, ci, docs, style, test, i18n are all `scopes` and not `types`?

This is what I am basically following the [Angular Commit Message Guildeline](https://gist.github.com/brianclements/841ea7bffdb01346392c), and I think most of their thoughts are suitable to ours.

> * I don't find a lot of value in knowing `fix` vs `feat`.  I think it's often pretty ambiguous (features can exist to "fix" old/missing code?)

I think if there are new functions/options/dungeon supports, I would treat it as a `feature`. If there are old code updating, such as "update actions id to match the new patch in-game" or "delete unusable monk lightning gauge because the new patch removed it", I prefer to treat it as a `fix` more than a `feature`, this code are actually "broken"/"bugged" in the new patch, isn't it?

If there is a PR contains both types of changes, I think `feat` is more important than `fix`, the latter one can be written as the description text of the PR.

Also, for `docs`/`style`/`i18n`/etc, they should be used if the PR contains the current type of changes ONLY. If there are some other changes, then the PR should be noted as `feat`/`fix` or something.

> * This feels slightly redundant with https://chris.beams.io/posts/git-commit/#imperative, where I find that a good PR title has an imperative verb to start (e.g. add/fix/refactor/remove) and so `fix(raidboss): fix thing` or `feat(oopsy): add thing` don't seem to add much more than `raidboss: fix thing` or `oopsy: add thing` and it takes up space in an already space-limited PR title

As my knowing, in this situation, you can just write `feat(raidboss): things` such as `feat(raidboss): Leviathan Unreal timeline` or `feat(raidboss): UCU triggers improvement` if there are a bunch of changes.

> * This is maybe a larger issue, but I am definitely not following semantic versioning for cactbot, mostly because I have never figured out what to really consider a breaking change (happy to discuss if people want to define this better?), and so semantic PR titles have less value because we're not doing semantic versioning

Mostly I was trying to make PR easy to be reviewed, so a semantic title would helpful in this situation. On the other hand, as Trim said, we can generate changelogs from semantic commit messages other than writing it manually.